joomla/joomla-cms
# [5.1] Replace deprecated substr() with substring()

### Summary of Changes

Javascript's `substr()` method has been deprecated for ages now (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) and this PR replaces it with `substring()`

### Testing Instructions

1. Inline Help should still work as expected.
2. Add a simple colour field to any XML file (`<field name="foo" type="color" label="Test" control="simple" />`) and ensure the field works as expected.
3. Ensure toggling the TineMCE presets within the Plugin Manager still works as expected.

I was unable to test the calendar changes as the code has no comments to assist developers, so this might need to be a code review instead.
